### DISCUSSION TITLE
Will there be new versions for java 8?

### DISCUSSION THREAD

**Discussion by thsandu** [awslabs/aws-java-nio-spi-for-s3#276] [General]
Hello,

as I stated in the general discussion thread, we plan to be using your cool library for integration tests with localstack. Unfortunately most of our projects are still using java 8, also our CI pipelines. 

This brings me to the question about your future plans. Will there be still version < 2.0.0 that support java 8? Or can we expect a 1.2.5 version in the near future?

If you are planning more java 8 versions, it would be a great help to have the issue #236 down merged to it. If you do not have the time, perhaps I could get some time to support here.

**Comment by markjschreiber** [comment]
Probably won't be future versions of 1.x but I'd be interested to know more about the use case for JDK 8. In most cases I don't think you need to develop with this library. Mostly with libs that hold java filesystem providers you just need to have them on your class path and the JVM will automagically use the lib when it sees an S3 path. So most applications will not need to include this lib as a dependency or even know it exists.

Sure, you will need JDK 11+ to use this as a provider but JDK 11 can run JDK 8 compiled bytecode so you can probably keep your application as JDK 8 and run it with 11 (along with this provider). Probably you can do the same with integration testing (use JDK 11 to run the tests) but JDK 8 to compile the applications that are being tested?

**Comment by thsandu** [comment]
Thanks for the quick reply @markjschreiber 

Our use case is, that some of our products are installed on-premises by our customers. And we know that almost all customers are still using JDK 8. So we decided to build that products with JDK 8.

I will try running the integration tests with JDK 11 and hopefully a possible integration with S3 will convince some of our customers to upgrade their machines to JDK 11.
